CMakeLists.txt

```diff
@@ -183,11 +183,14 @@ ELSE ()
             # -rdynamic
             )
 
-    if (CMAKE_BUILD_BITS EQUAL 32)
-        list(APPEND CXX_FLAGS "-m32")
-    else () #not-condition
-        list(APPEND CXX_FLAGS "-m64")
-    endif ()
+    option(BUILD_FOR_ARM "Build for arm64 platform" OFF)
```

Comment:
Why not detect the Arm platform and if it's Arm then skip this code instead of having a build option that means people building for Arm need to know to set this?